Sources/JWTKit/JWTHeader.swift

```diff
@@ -57,13 +57,24 @@ extension JWTHeader: Codable {
         self.x5c = try container.decodeIfPresent([String].self, forKey: .x5c)
 
         self.customFields = try container.allKeys
-            .filter { CodingKeys.allCases.contains($0) }
+            .filter { !CodingKeys.allCases.contains($0) }
             .reduce(into: [String: JWTHeaderField]()) { result, key in
                 result[key.stringValue] = try container.decode(JWTHeaderField.self, forKey: key)
             }
     }
 
     private enum CodingKeys: CodingKey, CaseIterable, Equatable {
```

Comment:
Don't conform to `CaseIterable` here - you're not actually fulfilling the protocol's requirements. You don't need to rename the property, just don't claim conformance to the protocol. (By declaring `CaseIterable` you're claiming that you're providing a complete and constant set of all possible enum cases, which isn't possible with a `String` as an associated value. You're also, more subtly, claiming that no two distinct cases are treated as equal, which is also not the case (for example, `.alg` and `.custom(name: "alg")` compare the same by raw value, but not by semantic - good thing you didn't let the compiler synthesize a `Hashable` conformance!). Violating the transitivity invariant is okay for `Equatable` in this context, but that's as far as I'd care to go.

Reply:
And while you're at it, you can change `allCases` to return a `Set<CodingKeys>` and then use `Set(container.allKeys).subtracting(CodingKeys.allCases)` above for an easy efficiency win.